### PR TITLE
Prevent accidental misuse of Sonar token

### LINKED ISSUE #101: SonarCloud doesn't seem to validate token against org and project name
In some cases a valid `SONAR_TOKEN` can be used to upload analysis report to another Sonar project. It seems that the token is not validated against `sonar.organization` and `sonar.projectKey`. This needs further investigation. 

One of the options is to load both configuration settings `sonar.organization` and `sonar.projectKey` from environment variables.

### PR DESCRIPTION
**Pull request by stefaniuk**
## Description

Prevent accidental misuse of Sonar token by forcing to store organisation key and project key as GitHub secrets.

- Fixes #101 

## Context

Sonar token is a user token.

## Type of changes

- [x] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: .github/actions/perform-static-analysis/action.yaml [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+  sonar_organisation_key:
+    description: "Sonar organisation key, used to identify the project"
+    required: false
+  sonar_project_key:
+    description: "Sonar project key, used to identify the project"
+    required: false
   sonar_token:
-    description: "Sonar token API key"
+    description: "Sonar token, the API key"
     required: false
```
Comment (jacobgill1) [inline .github/actions/perform-static-analysis/action.yaml:12]: Could the `secrets:` key be used here instead? I'm not too sure on the differences but I believe it helps GitHub in identifying the value as secret and obscuring it in outputs etc

**Review by jacobgill1 — COMMENTED** [review]
Looks good! I've just left the one question.

File: .github/actions/perform-static-analysis/action.yaml (hunk 1 above)
Comment (stefaniuk) [reply .github/actions/perform-static-analysis/action.yaml:12]: Good question! As this is a composite action it appears that even a secret has to be passed as an input. However, it is still reduced from the output as the calling workflow is aware it shouldn't be display in plaintext:

image

Comment (jacobgill1) [reply .github/actions/perform-static-analysis/action.yaml:12]: Makes sense!

**Review by jacobgill1 — APPROVED** [review]
LGTM

**Pull request merged**